12-Glossary.md

```diff
@@ -1 +1,40 @@
-# Glossary
\ No newline at end of file
+# Glossary
+
+This section contains the most important domain and technical terms used when discussing the system.
+The goal is for all consumers of the architecture documentation to have an identical understanding of the terms and not use synonyms and homonyms.
+
+## Stakeholder
+
+All persons, roles or organizations that:
+- will use the system
+- should know the architecture
+- have to work with the architecture or with code
+- need the documentation of the architecture for their work
+- have to come up with decisions about the system or its development
+
+## Use case
+
+A textual description of the functionality provided by the system that captures actor-system interaction. It specifies how a user interacts with a system and how the system responds to the user actions.
+
+## Component
+
+A component is a software unit with a well-defined interface and explicitly specified dependencies. A software component can be as small as a block of reusable code, or it can be as big as an entire application depending on the level of granularity a system is being looked at.
+
+## MediaWiki Extension
+
+Extensions are software components that administrators can add (or remove) to their MediaWiki application in order to customize how it looks and works.
+Examples: WikibaseClient, UniversalLanguageSelector, OAuth
```

Comment:
What do you think about completely removing any text for terms and concepts that are well-defined elsewhere like MediaWiki concepts and Wikibase data model terminology, and only link to the respective site?

```suggestion
[https://www.mediawiki.org/wiki/Manual:Extensions](https://www.mediawiki.org/wiki/Manual:Extensions)
```

... and the same for sitelinks and interwiki links.

Reply:
Ok, after a quick chat we agreed that brief descriptions are useful. That's fine with me, too!